R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
+                                 priorValues = hyperparameterList[[hyperparameterSet]],
+                                 alternativeRestriction = alternativeRestriction,
+                                 alpha = alpha, beta = beta,
+                                 delta = deltaGenerating,
+                                 deltaDesign = deltaDesign,
+                                 M = M,
+                                 maxSimStoptime = maxSimStoptime,
+                                 gridSize = thetaAgridSize)[["worstCaseQuantile"]]
+      resultDataFrame <- rbind(resultDataFrame,
+                               data.frame(hyperparameters = hyperparameterSet,
+                                          delta = deltaGenerating,
+                                          worstCaseQuantile = worstCase)
+                               )
     }
   }
 
-  if(makePlot){
-    graphics::hist(n.final, breaks = na.max+nb.max, xlim = c(0, max(n.final)), xlab = "n collected",
-                   main = bquote(~"Mean 1" == .(parametersDataGeneratingDistribution[1]) ~"," ~"Mean 2" == .(parametersDataGeneratingDistribution[2])  ~"," ~"alpha" == ~.(safeDesign$alpha)))
-
+  #now we know the worst case quantiles, retrieve expected stopping times
+  message(paste("Retrieving all expected stopping times given the worst case stopping times"))
+  for(i in 1:nrow(resultDataFrame)){
+    resultDataFrame[i,"expected"] <- simulateWorstCaseQuantile(na = na, nb = nb,
+                                          priorValues = hyperparameterList[[resultDataFrame[i, "hyperparameters"]]],
+                                          alternativeRestriction = alternativeRestriction,
+                                          alpha = alpha, beta = 0,
+                                          delta = resultDataFrame[i, "delta"],
+                                          deltaDesign = deltaDesign,
+                                          M = M,
+                                          #now we stop, each experiment, maximally at the worst case stoptime for 80% power
+                                          maxSimStoptime = ceiling(resultDataFrame[i,"worstCaseQuantile"]),
+                                          gridSize = thetaAgridSize,
+                                          #and we calculate the expectation instead of a (1-b) quantile
+                                          expectedStopTime = TRUE)[["worstCaseQuantile"]]
   }
-  return(list(allN = n.final,
-              rejected = rejected,
-              s_values = s.final,
-              allRejectedN = n.final[rejected]))
+
+  simResult <- list(simdata = resultDataFrame,
+                    alpha = alpha,
+                    beta = beta,
+                    deltaDesign = deltaDesign,
+                    restriction = alternativeRestriction,
+                    hyperparameters = hyperparameterList
+                    )
+  class(simResult) <- "safe2x2Sim"
```

Comment:
Willen we misschien een generieke `safeSim` klasse maken? Ik dacht dat dat niet nodig zou zijn, omdat we dit niet heel vaak doen.

Reply:
ik zie daar nu niet direct het voordeel van; mijn plotfunctie is wel heel anders dan wat jij wil doet. Ik laat ze meerdere prior opties invoeren en die worden dan vergeleken.